File: April 2022/April_24.cpp
class UndergroundSystem {
public:
    map<int, pair<string, int> > arrival;
    map<string, pair<int, int> > mp;
    
    UndergroundSystem() {
        arrival.clear();
        mp.clear();
    }
    
    void checkIn(int id, string stationName, int t) {
        arrival[id] = make_pair(stationName, t);
    }
    
    void checkOut(int id, string stationName, int t) {
        auto p = arrival[id];
        string key = p.first + " " + stationName;
        if(mp.find(key) == mp.end())
            mp[key] = make_pair(t - p.second, 1);
        else{
            auto p2 = mp[key];
            mp[key] = make_pair(p2.first + t - p.second, p2.second + 1);
        }
    }
    
    double getAverageTime(string startStation, string endStation) {
        auto p = mp[startStation + " " + endStation];
        return 1.0 * p.first / p.second;
    }
};

/**
 * Your UndergroundSystem object will be instantiated and called as such:
 * UndergroundSystem* obj = new UndergroundSystem();
 * obj->checkIn(id,stationName,t);
 * obj->checkOut(id,stationName,t);
 * double param_3 = obj->getAverageTime(startStation,endStation);
 */

File: April 2022/April_24.java
class Pair{
    String name;
    int t;
    Pair(String name, int t){
        this.name = name;
        this.t = t;
    }
}

class Pair2{
    int t, f;
    Pair2(int t, int f){
        this.t = t;
        this.f = f;
    }
}

class UndergroundSystem {
    Map<Integer, Pair> map;
    Map<String, Pair2> map2;

    public UndergroundSystem() {
        map = new HashMap<>();
        map2 = new HashMap<>();
    }
    
    public void checkIn(int id, String stationName, int t) {
        map.put(id, new Pair(stationName, t));
    }
    
    public void checkOut(int id, String stationName, int t) {
        Pair p = map.get(id);
        String key = p.name + " " + stationName;
        if(map2.containsKey(key)){
            Pair2 p2 = map2.get(key);
            map2.put(key, new Pair2(p2.t + t - p.t, p2.f + 1));
        } else
            map2.put(key, new Pair2(t - p.t, 1));
    }
    
    public double getAverageTime(String startStation, String endStation) {
        Pair2 p = map2.get(startStation + " " + endStation);
        return 1.0 * p.t / p.f;
    }
}

/**
 * Your UndergroundSystem object will be instantiated and called as such:
 * UndergroundSystem obj = new UndergroundSystem();
 * obj.checkIn(id,stationName,t);
 * obj.checkOut(id,stationName,t);
 * double param_3 = obj.getAverageTime(startStation,endStation);
 */
